### ISSUE TITLE
magento 2.4.6 cron issues reported in log file

### ISSUE THREAD

**Issue by medigeek** [calcurates/module-magento#6]
```
==> /var/www/html/var/log/debug.log <==
[2023-03-24T17:54:06.575487+00:00] main.CRITICAL: Error: Class "Zend_Json" not found in /var/www/html/vendor/calcurates/module-magento/Client/Command/GetShippingOptionsCommand.php:100
Stack trace:
#0 /var/www/html/vendor/calcurates/module-magento/Client/Command/GetAllShippingOptionsCommand.php(40): Calcurates\ModuleMagento\Client\Command\GetShippingOptionsCommand->get()
#1 /var/www/html/vendor/calcurates/module-magento/Model/Carrier.php(320): Calcurates\ModuleMagento\Client\Command\GetAllShippingOptionsCommand->getShippingOptions()
#2 /var/www/html/vendor/mageworx/module-ordersgrid/Model/Config/Source/ShippingMethods.php(62): Calcurates\ModuleMagento\Model\Carrier->getAllowedMethods()
#3 /var/www/html/vendor/mageworx/module-ordersgrid/Model/ResourceModel/GridAdditionalData.php(174): MageWorx\OrdersGrid\Model\Config\Source\ShippingMethods->toOptionArray()
#4 /var/www/html/vendor/mageworx/module-ordersgrid/Model/ResourceModel/GridAdditionalData.php(221): MageWorx\OrdersGrid\Model\ResourceModel\GridAdditionalData->getAllShippingMethods()
#5 /var/www/html/vendor/mageworx/module-ordersgrid/Model/Indexer/AdditionalData.php(41): MageWorx\OrdersGrid\Model\ResourceModel\GridAdditionalData->rebuildIndexTable()
#6 /var/www/html/generated/code/MageWorx/OrdersGrid/Model/Indexer/AdditionalData/Interceptor.php(23): MageWorx\OrdersGrid\Model\Indexer\AdditionalData->execute()
#7 /var/www/html/vendor/mageworx/module-ordersgrid/Model/Indexer/AdditionalData.php(49): MageWorx\OrdersGrid\Model\Indexer\AdditionalData\Interceptor->execute()
#8 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(58): MageWorx\OrdersGrid\Model\Indexer\AdditionalData->executeFull()
#9 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(138): MageWorx\OrdersGrid\Model\Indexer\AdditionalData\Interceptor->___callParent()
#10 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(153): MageWorx\OrdersGrid\Model\Indexer\AdditionalData\Interceptor->Magento\Framework\Interception\{closure}()
#11 /var/www/html/generated/code/MageWorx/OrdersGrid/Model/Indexer/AdditionalData/Interceptor.php(32): MageWorx\OrdersGrid\Model\Indexer\AdditionalData\Interceptor->___callPlugins()
#12 /var/www/html/vendor/magento/module-indexer/Model/Indexer.php(443): MageWorx\OrdersGrid\Model\Indexer\AdditionalData\Interceptor->executeFull()
#13 /var/www/html/vendor/magento/module-indexer/Model/Indexer/DependencyDecorator.php(268): Magento\Indexer\Model\Indexer->reindexAll()
#14 /var/www/html/vendor/magento/module-indexer/Model/Processor.php(88): Magento\Indexer\Model\Indexer\DependencyDecorator->reindexAll()
#15 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Indexer\Model\Processor->reindexAllInvalid()
#16 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Indexer\Model\Processor\Interceptor->___callParent()
#17 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Indexer\Model\Processor\Interceptor->Magento\Framework\Interception\{closure}()
#18 /var/www/html/generated/code/Magento/Indexer/Model/Processor/Interceptor.php(23): Magento\Indexer\Model\Processor\Interceptor->___callPlugins()
#19 /var/www/html/vendor/magento/module-indexer/Cron/ReindexAllInvalid.php(31): Magento\Indexer\Model\Processor\Interceptor->reindexAllInvalid()
#20 [internal function]: Magento\Indexer\Cron\ReindexAllInvalid->execute()
#21 /var/www/html/vendor/magento/module-cron/Observer/ProcessCronQueueObserver.php(368): call_user_func_array()
#22 /var/www/html/vendor/magento/module-cron/Observer/ProcessCronQueueObserver.php(879): Magento\Cron\Observer\ProcessCronQueueObserver->_runJob()
#23 /var/www/html/vendor/magento/module-cron/Observer/ProcessCronQueueObserver.php(840): Magento\Cron\Observer\ProcessCronQueueObserver->tryRunJob()
#24 /var/www/html/vendor/magento/module-cron/Observer/ProcessCronQueueObserver.php(280): Magento\Cron\Observer\ProcessCronQueueObserver->processPendingJobs()
#25 /var/www/html/vendor/magento/framework/Event/Invoker/InvokerDefault.php(88): Magento\Cron\Observer\ProcessCronQueueObserver->execute()
#26 /var/www/html/vendor/magento/framework/Event/Invoker/InvokerDefault.php(74): Magento\Framework\Event\Invoker\InvokerDefault->_callObserverMethod()
#27 /var/www/html/vendor/magento/framework/Event/Manager.php(65): Magento\Framework\Event\Invoker\InvokerDefault->dispatch()
#28 /var/www/html/generated/code/Magento/Framework/Event/Manager/Proxy.php(95): Magento\Framework\Event\Manager->dispatch()
#29 /var/www/html/vendor/magento/framework/App/Cron.php(86): Magento\Framework\Event\Manager\Proxy->dispatch()
#30 /var/www/html/vendor/magento/module-cron/Console/Command/CronCommand.php(126): Magento\Framework\App\Cron->launch()
#31 /var/www/html/vendor/symfony/console/Command/Command.php(298): Magento\Cron\Console\Command\CronCommand->execute()
#32 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(58): Symfony\Component\Console\Command\Command->run()
#33 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Cron\Console\Command\CronCommand\Interceptor->___callParent()
#34 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Cron\Console\Command\CronCommand\Interceptor->Magento\Framework\Interception\{closure}()
#35 /var/www/html/generated/code/Magento/Cron/Console/Command/CronCommand/Interceptor.php(23): Magento\Cron\Console\Command\CronCommand\Interceptor->___callPlugins()
#36 /var/www/html/vendor/symfony/console/Application.php(1040): Magento\Cron\Console\Command\CronCommand\Interceptor->run()
#37 /var/www/html/vendor/symfony/console/Application.php(301): Symfony\Component\Console\Application->doRunCommand()
#38 /var/www/html/vendor/magento/framework/Console/Cli.php(116): Symfony\Component\Console\Application->doRun()
#39 /var/www/html/vendor/symfony/console/Application.php(171): Magento\Framework\Console\Cli->doRun()
#40 /var/www/html/bin/magento(23): Symfony\Component\Console\Application->run()
#41 {main}

Next RuntimeException: Error when running a cron job: Class "Zend_Json" not found in /var/www/html/vendor/magento/module-cron/Observer/ProcessCronQueueObserver.php:380
Stack trace:
#0 /var/www/html/vendor/magento/module-cron/Observer/ProcessCronQueueObserver.php(879): Magento\Cron\Observer\ProcessCronQueueObserver->_runJob()
#1 /var/www/html/vendor/magento/module-cron/Observer/ProcessCronQueueObserver.php(840): Magento\Cron\Observer\ProcessCronQueueObserver->tryRunJob()
#2 /var/www/html/vendor/magento/module-cron/Observer/ProcessCronQueueObserver.php(280): Magento\Cron\Observer\ProcessCronQueueObserver->processPendingJobs()
#3 /var/www/html/vendor/magento/framework/Event/Invoker/InvokerDefault.php(88): Magento\Cron\Observer\ProcessCronQueueObserver->execute()
#4 /var/www/html/vendor/magento/framework/Event/Invoker/InvokerDefault.php(74): Magento\Framework\Event\Invoker\InvokerDefault->_callObserverMethod()
#5 /var/www/html/vendor/magento/framework/Event/Manager.php(65): Magento\Framework\Event\Invoker\InvokerDefault->dispatch()
#6 /var/www/html/generated/code/Magento/Framework/Event/Manager/Proxy.php(95): Magento\Framework\Event\Manager->dispatch()
#7 /var/www/html/vendor/magento/framework/App/Cron.php(86): Magento\Framework\Event\Manager\Proxy->dispatch()
#8 /var/www/html/vendor/magento/module-cron/Console/Command/CronCommand.php(126): Magento\Framework\App\Cron->launch()
#9 /var/www/html/vendor/symfony/console/Command/Command.php(298): Magento\Cron\Console\Command\CronCommand->execute()
#10 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(58): Symfony\Component\Console\Command\Command->run()
#11 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Cron\Console\Command\CronCommand\Interceptor->___callParent()
#12 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Cron\Console\Command\CronCommand\Interceptor->Magento\Framework\Interception\{closure}()
#13 /var/www/html/generated/code/Magento/Cron/Console/Command/CronCommand/Interceptor.php(23): Magento\Cron\Console\Command\CronCommand\Interceptor->___callPlugins()
#14 /var/www/html/vendor/symfony/console/Application.php(1040): Magento\Cron\Console\Command\CronCommand\Interceptor->run()
#15 /var/www/html/vendor/symfony/console/Application.php(301): Symfony\Component\Console\Application->doRunCommand()
#16 /var/www/html/vendor/magento/framework/Console/Cli.php(116): Symfony\Component\Console\Application->doRun()
#17 /var/www/html/vendor/symfony/console/Application.php(171): Magento\Framework\Console\Cli->doRun()
#18 /var/www/html/bin/magento(23): Symfony\Component\Console\Application->run()
#19 {main} [] []
```

**Comment by medigeek** [comment]
Looks like this is happening because of the removal of Zend_* libraries:

https://experienceleague.adobe.com/docs/commerce-operations/release/notes/adobe-commerce/2-4-6.html

> Zend framework (ZF1) components that have reached end of life have been removed from the codebase. See [Backward-incompatible changes highlights](https://developer-stage.adobe.com/commerce/php/development/backward-incompatible-changes/highlights/).

https://developer-stage.adobe.com/commerce/php/development/backward-incompatible-changes/highlights/#246

> 2.4.6
> The following major backward-incompatible changes were introduced in the 2.4.6 Adobe Commerce and Magento Open Source releases:
> 
> New system configuration for customer segments
> New system configuration for limiting products in grid
> New system configuration for OpenSearch module
> Symfony dependencies upgraded to latest LTS version
> Zend_Filter replaced with laminas-filter
> Zend_HTTP replaced with laminas-http
> Zend_Validate replaced with laminas-validator

Other modules have had similar issues with Zend_Date in the cron for example:
https://github.com/Smile-SA/elasticsuite/issues/2818#issuecomment-1380999616

About Zend_Json there is a replacement available:
https://magento.stackexchange.com/questions/365425/zend-replacement-in-2-4-6

> Zend usage deprecated since 2.4.6 Now you can use Laminas. Usage: replace `Zend_Json` with `\Laminas\Json\Json` Be sure that you have installed `laminas/laminas-serializer`

I'll leave it to you from here on though. :)

**Comment by Gemorroj** [comment]
@medigeek this issue has been fixed in [tag/v1.46.2](../releases/tag/v1.46.2)
